client/my-sites/plugins/plugins-discovery-page/index.jsx

```diff
@@ -0,0 +1,207 @@
+import {
+	FEATURE_INSTALL_PLUGINS,
+	findFirstSimilarPlanKey,
+	getPlan,
+	isBlogger,
+	isPersonal,
+	isPremium,
+	TYPE_BUSINESS,
+	WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS,
+} from '@automattic/calypso-products';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import UpsellNudge from 'calypso/blocks/upsell-nudge';
+import { isEligibleForProPlan } from 'calypso/my-sites/plans-comparison';
+import { isCompatiblePlugin } from 'calypso/my-sites/plugins/plugin-compatibility';
+import getPlansForFeature from 'calypso/state/selectors/get-plans-for-feature';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
+import './style.scss';
+import SingleListView from '../plugins-browser/single-list-view';
+import usePlugins from '../use-plugins';
+
+/**
+ * Module variables
+ */
+const SHORT_LIST_LENGTH = 6;
+
+const UpgradeNudge = ( { selectedSite, sitePlan, isVip, jetpackNonAtomic, siteSlug } ) => {
+	const translate = useTranslate();
+	const eligibleForProPlan = useSelector( ( state ) =>
+		isEligibleForProPlan( state, selectedSite?.ID )
+	);
+
+	if ( ! selectedSite?.ID || ! sitePlan || isVip || jetpackNonAtomic ) {
+		return null;
+	}
+	const isLegacyPlan = isBlogger( sitePlan ) || isPersonal( sitePlan ) || isPremium( sitePlan );
+	const checkoutPlan = eligibleForProPlan && ! isLegacyPlan ? 'pro' : 'business';
+	const bannerURL = `/checkout/${ siteSlug }/${ checkoutPlan }`;
+	const plan = findFirstSimilarPlanKey( sitePlan.product_slug, {
+		type: TYPE_BUSINESS,
+	} );
+
+	const title =
+		eligibleForProPlan && ! isLegacyPlan
+			? translate( 'Upgrade to the Pro plan to install plugins.' )
+			: translate( 'Upgrade to the Business plan to install plugins.' );
+
+	return (
+		<UpsellNudge
+			event="calypso_plugins_browser_upgrade_nudge"
+			showIcon={ true }
+			href={ bannerURL }
+			feature={ FEATURE_INSTALL_PLUGINS }
+			plan={ plan }
+			title={ title }
+		/>
+	);
+};
+
+const UpgradeNudgePaid = ( props ) => {
```

Comment:
I wonder if in the future we can do something to mix both `UpgradeNudge` and `UpgradeNudgePaid` in one

Reply:
Sounds good. I will do it on this PR to avoid to create a new one.

Reply:
@lupus2k I suggest tackling this one as a follow-up task

Reply:
I followed up in [this issue](https://github.com/Automattic/wp-calypso/issues/66327)